Rdm 4177 event selector new contract

adnanakgun opened the pull request:
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDM-4177

### Change description ###

Moved error management from component to consumer. Some refactoring too. 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

adnanakgun commented:
@andrewfolga In line with what Nayab said regarding updated PR's on Slack, do you want to re-review?

andrewfolga commented:
QA testing this - please make sure you test the callback error/warning scenarios to make sure the proper error banner is displayed and button disabling logic works.

adnanakgun commented:
@andrewfolga I pushed another commit to this branch. Noticed it was needed while building the demo.
file: event-trigger.component.ts
Changed:
`if (changes.triggers) {`
to:
`if (changes.triggers && changes.triggers.currentValue) {`

file: event-trigger.component.html
Changed:
`<form *ngIf="triggers.length" class="event-trigger" (ngSubmit)="triggerSubmit()" [formGroup]="triggerForm">`
to:
`<form *ngIf="triggers" class="event-trigger" (ngSubmit)="triggerSubmit()" [formGroup]="triggerForm">`
